grpc-sys/build.rs

```diff
@@ -234,7 +237,78 @@ fn get_env(name: &str) -> Option<String> {
     }
 }
 
+/**
+ * Generate the bindings to C-core
+ */
+fn bindgen_grpc() {
+    let mut config = config_from_headers();
+
+    if get_env("CARGO_CFG_TARGET_OS").map_or(false, |s| s == "windows") {
+        // At lease win7
+        config = config.clang_arg("-D _WIN32_WINNT=0x0700");
+    }
+
+    let bindings = config
+        .clang_arg("-I./grpc/include")
+        .clang_arg("-DGRPC_SYS_SECURE")
+        .clang_arg("-std=c++11")
+        .whitelist_function(r"\bgrpc_.*")
+        .whitelist_function(r"\bgpr_.*")
+        .whitelist_function(r"\bgrpcwrap_.*")
+        .rustified_enum(r"\bgrpc_.*")
+        .rustified_enum(r"\bgpr_.*")
+        .no_copy("grpc_slice")
+        .generate()
+        .expect("Unable to generate grpc bindings");
+
+    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
+    bindings
+        .write_to_file(out_path.join("grpc-bindings.rs"))
+        .expect("Couldn't write bindings!");
+}
+
+/**
+ * Use grep to search header files with API interface in the
+ * include/grpc directory and add these header files to bindgen's config
+ */
+fn config_from_headers() -> bindgen::Builder {
+    let mut config = bindgen::Builder::default();
+    let mut headers: Vec<String> = Vec::new();
+    let matcher = RegexMatcher::new("GRPCAPI|GPRAPI").expect("can't resolve pattern");
+
+    for result in WalkDir::new(Path::new("./grpc/include")) {
+        let dent = match result {
+            Ok(dent) => dent,
+            Err(err) => {
+                eprintln!("{}", err);
+                continue;
+            }
+        };
+        if !dent.file_type().is_file() {
+            continue;
+        }
+        let result = grep::searcher::Searcher::new().search_path(
```

Comment:
Can you use pure string comparison instead?

Reply:
Did not fully understand...